src/include/storage/storage_defs.h

```diff
@@ -176,30 +178,48 @@ class VarlenEntry {
    * @param size length of the varlen content, in bytes (no C-style nul-terminator)
    * @param gathered whether the varlen entry's content pointer is part of a large buffer (for arrow-compatibility),
    *                 which means it cannot be deallocated by itself.
+   * @param compressed whether the varlen entry is stored using dictionnary compression
    */
-  VarlenEntry(byte *content, uint32_t size, bool gathered)
-      // the sign bit on size is used to store the "gathered" attribute, so we mask it off on size depending on that.
-      : size_(size | (gathered ? INT32_MIN : 0)), content_(content) {}
+  VarlenEntry(byte *content, uint32_t size, bool gathered, bool compressed)
```

Comment:
Two boolean arguments are confusing when reading code. Let's take in an enum for the constructor so it's explicit. e.g. VarlenEntry(bytes, VarlenEntryType::COMPRESSED)